### DISCUSSION TITLE
Hosting in Azure difficulty

### DISCUSSION THREAD

**Discussion by eruantiendraugole** [lloesche/valheim-server-docker#137] [Q&A] [answered]
I am trying to host on Azure, have uploaded and volume mounted to azure file share, however I cannot see the server by the public IP and port. have included the container JSON below, hoping someone can see something stupid I have done.      

      {
          "properties": {
              "sku": "Standard",
              "provisioningState": "Succeeded",
              "containers": [
                  {
                      "name": "valheim",
                      "properties": {
                          "image": "lloesche/valheim-server:main",
                          "ports": [
                              {
                                  "protocol": "UDP",
                                  "port": 2456
                              },
                              {
                                  "protocol": "UDP",
                                  "port": 2457
                              },
                              {
                                  "protocol": "UDP",
                                  "port": 2458
                              }
                          ],
                          "environmentVariables": [
                              {
                                  "name": "SERVER_PASS",
                                  "value": "xxxxx"
                              },
                              {
                                  "name": "BACKUPS_MAX_AGE",
                                  "value": "10"
                              },
                              {
                                  "name": "SERVER_NAME",
                                  "value": "xxxxx"
                              },
                              {
                                  "name": "SERVER_PORT",
                                  "value": "2456"
                              },
                              {
                                  "name": "WORLD_NAME",
                                  "value": "xxxx"
                              }
                          ],
                          "instanceView": {
                              "restartCount": 38,
                              "currentState": {
                                  "state": "Running",
                                  "startTime": "2021-03-02T11:23:34Z",
                                  "detailStatus": ""
                              },
                              "previousState": {
                                  "state": "Terminated",
                                  "startTime": "2021-03-02T11:00:17Z",
                                  "exitCode": 137,
                                  "finishTime": "2021-03-02T11:23:29Z",
                                  "detailStatus": "OOMKilled"
                              },
                              "events": [
                                  {
                                      "count": 3,
                                      "firstTimestamp": "2021-03-01T18:13:35Z",
                                      "lastTimestamp": "2021-03-01T19:15:09Z",
                                      "name": "Pulling",
                                      "message": "pulling image \"lloesche/valheim-server:main\"",
                                      "type": "Normal"
                                  },
                                  {
                                      "count": 2,
                                      "firstTimestamp": "2021-03-01T18:14:13Z",
                                      "lastTimestamp": "2021-03-01T18:45:14Z",
                                      "name": "Pulled",
                                      "message": "Successfully pulled image \"lloesche/valheim-server:main\"",
                                      "type": "Normal"
                                  },
                                  {
                                      "count": 2,
                                      "firstTimestamp": "2021-03-01T18:14:21Z",
                                      "lastTimestamp": "2021-03-01T18:45:14Z",
                                      "name": "Created",
                                      "message": "Created container",
                                      "type": "Normal"
                                  },
                                  {
                                      "count": 2,
                                      "firstTimestamp": "2021-03-01T18:14:21Z",
                                      "lastTimestamp": "2021-03-01T18:45:15Z",
                                      "name": "Started",
                                      "message": "Started container",
                                      "type": "Normal"
                                  },
                                  {
                                      "count": 38,
                                      "firstTimestamp": "2021-03-01T18:45:08Z",
                                      "lastTimestamp": "2021-03-02T11:23:29Z",
                                      "name": "Killing",
                                      "message": "Killing container with id docker://valheim:Need to kill Pod",
                                      "type": "Normal"
                                  }
                              ]
                          },
                          "resources": {
                              "requests": {
                                  "memoryInGB": 1,
                                  "cpu": 1
                              },
                              "limits": {
                                  "memoryInGB": 1,
                                  "cpu": 1
                              }
                          },
                          "volumeMounts": [
                              {
                                  "name": "azurefileshare",
                                  "mountPath": "/config"
                              },
                              {
                                  "name": "azurefilesharedata",
                                  "mountPath": "/opt/valheim"
                              }
                          ]
                      }
                  }
              ],
              "initContainers": [],
              "restartPolicy": "Always",
              "ipAddress": {
                  "ports": [
                      {
                          "protocol": "UDP",
                          "port": 2456
                      },
                      {
                          "protocol": "UDP",
                          "port": 2457
                      },
                      {
                          "protocol": "UDP",
                          "port": 2458
                      }
                  ],
                  "ip": "51.11.173.150",
                  "type": "Public"
              },
              "osType": "Linux",
              "volumes": [
                  {
                      "name": "azurefileshare",
                      "azureFile": {
                          "shareName": "valheimvalheim-serverconfig",
                          "readOnly": false,
                          "storageAccountName": "valheimnerdzstorage"
                      }
                  },
                  {
                      "name": "azurefilesharedata",
                      "azureFile": {
                          "shareName": "valheimvalheim-serverdata",
                          "readOnly": false,
                          "storageAccountName": "valheimnerdzstorage"
                      }
                  }
              ],
              "instanceView": {
                  "events": [
                      {
                          "count": 1,
                          "firstTimestamp": "2021-03-01T18:13:32Z",
                          "lastTimestamp": "2021-03-01T18:13:32Z",
                          "name": "SuccessfulMountAzureFileVolume",
                          "message": "Successfully mounted Azure File Volume.",
                          "type": "Normal"
                      },
                      {
                          "count": 1,
                          "firstTimestamp": "2021-03-01T18:13:34Z",
                          "lastTimestamp": "2021-03-01T18:13:34Z",
                          "name": "SuccessfulMountAzureFileVolume",
                          "message": "Successfully mounted Azure File Volume.",
                          "type": "Normal"
                      }
                  ],
                  "state": "Running"
              }
          },
          "id": "/subscriptions/606d1fbf-d00f-40c7-87f9-8b5837c6249f/resourceGroups/valheim/providers/Microsoft.ContainerInstance/containerGroups/dockervalheim",
          "name": "dockervalheim",
          "type": "Microsoft.ContainerInstance/containerGroups",
          "location": "uksouth",
          "tags": {
              "docker-compose-application": "docker-compose-application"
          }
      }

**Comment by lloesche** [comment, marked as answer]
I have been completely unable to get it running on Azure EU with the following error https://github.com/lloesche/valheim-server-docker/issues/120
The container host kernel does not support running 32 bit ELF binaries but `steamcmd` only comes as 32 bit.

Some folks over in US locations reported having luck with running 32 bit binaries on their container hosts.
